### DISCUSSION TITLE
Server Scripting tool (findings and fixes) and LabVIEW clients creating

### DISCUSSION THREAD

**Discussion by lbaute** [ni/grpc-device#296] [General]
Hello! 
I've been trying to use what is on the repo, but seems like there are some files missing, like labview_grpc_generator.dll and  folder where this one resides, and other files
I was able to find it (I don't remember where) but when trying to call it again it was not there on the main repo.

On the other hand when running to generate servers with the scripting tools some proto files have definitions that make the generation break, so I've changed them by hand to allow the tool run... like bytes for string, fixed64 for uint64, sint32 for int32... I've find them here.
Reference: https://developers.google.com/protocol-buffers/docs/proto3#scalar

I was able to build Servers and LabVIEW clients for Scope, NI switch, would you like to talk more with youy, 

On the other hand I would appreciate if you'd be willing to do a intro to gRPC and the approach NI is having towards it, Thanks I was redirected from tech support here

Id love to gather together, thanks for your time

**Comment by harsha-bhushan** [comment]
Hi, Thanks for reaching out. I think you are referring to another repo - [grpc-labview ](https://github.com/ni/grpc-labview) (which has the dll you mention)
We have not yet added support for LabVIEW clients for the drivers. The script provided by the [grpc-labview](https://github.com/ni/grpc-labview) repo doesn't generate LabVIEW clients yet, but only generates the server. 

Could you elaborate on how were able to build the LabVIEW clients for NI Scope & NI Switch?
Also is there a reason why you want to generate the server in LabVIEW rather than use the server binaries released in this repo, (with the driver support already available)? 

It is not recommended to change the data types in the driver proto files as that may lead to incorrect behavior of the functions. 
Could you share the name of the person from tech support whom you were in touch with (so that I can loop them in)?

**Comment by lbaute** [comment]
Hi , To generate servers I need a proto file which was taken from ni/grpc-device repo. 
But when I run some of them crash due to dll outputting the following error:  

Error 5020 occurred at gRPC Scripting Tools.lvlib:Error 5020.vi
Possible reason(s):
Proto file import - get field info

Error 5015 occurred at gRPC Scripting Tools.lvlib:Error 5015.vi
Possible reason(s):
Proto file import - messages ref error

These errors, are coming from 
Library: labview_grpc_generator.dll
Function Name:  int32_t LVGetMessages(uintptr_t protoDescription, Array1DUintptr_t ***messages);
and Function Name: int32_t LVGetServices(uintptr_t protoDescription, Array1DUintptr_t ***services);

If I fixed the types mentioned above it runs without errors
Is this something expected? or am I missing something.

Support person is Liam MacShane
